### PR TITLE
CanUseTopic() refactor.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request 
Supposedly the second half of #54665 that I decided to atomize out.

## Why It's Good For The Game
Streamlining `canUseTopic` (and 'default_can_use_topic' because tgui) and making dexterity checks a bit more consistent to all mobs.
**Now tested and working.**

## Changelog
:cl:
fix: Jungle fever monkeys will now pass canUseTopic() dexterity checks and be able to use tgui machineries.
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
> Still have to test it, but I'm confident it works.

😏

**Comment by LemonInTheDark** [comment]
Test it 🗞️

File: code/modules/tgui/states.dm [hunk 1]
```diff
@@ -89,22 +89,6 @@
...
- * Check the distance for a living mob.
- * Really only used for checks outside the context of a mob.
- * Otherwise, use shared_living_ui_distance().
- *
- * required src_object The object which owns the UI.
- * required user mob The mob who opened/is using the UI.
- *
- * return UI_state The state of the UI.
- */
-/atom/proc/contents_ui_distance(src_object, mob/living/user)
```
Comment (Ghommie) [inline code/modules/tgui/states.dm:104]: Only ever used in one place and it's really just proc overhead.

File: code/__DEFINES/misc.dm [hunk 2]
```diff
@@ -73,8 +73,9 @@
 
 #define BE_CLOSE TRUE		//in the case of a silicon, to select if they need to be next to the atom
 #define NO_DEXTERITY TRUE	//if other mobs (monkeys, aliens, etc) can use this // I had to change 20+ files because some non-dnd-playing fuckchumbis can't spell "dexterity"
-#define NO_TK TRUE
-#define FLOOR_OKAY TRUE     // if you can use it while resting
+#define NO_TK TRUE			// if you can't use it from a distance with telekinesis
+#define FLOOR_OKAY TRUE		// if you can use it while resting
+#define NEED_HANDS TRUE		// if the mob needs an usable hand, regardless of dexterity.
```
Comment (AnturK) [inline code/__DEFINES/misc.dm:78]: Expanding on this nonsense is causing me physical pain.

Comment (Ghommie) [reply code/__DEFINES/misc.dm:78]: I just didn't feel like defying that senseless bandwagon, it's been scratching me an itch too tbh.

**Pull request merged**